src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/SpanEditHandler.cs

```diff
@@ -15,29 +15,21 @@ internal class SpanEditHandler
 {
     private static readonly int TypeHashCode = typeof(SpanEditHandler).GetHashCode();
 
-    public SpanEditHandler(Func<string, IEnumerable<Syntax.InternalSyntax.SyntaxToken>> tokenizer)
-        : this(tokenizer, AcceptedCharactersInternal.Any)
-    {
-    }
-
-    public SpanEditHandler(Func<string, IEnumerable<Syntax.InternalSyntax.SyntaxToken>> tokenizer, AcceptedCharactersInternal accepted)
-    {
-        AcceptedCharacters = accepted;
-        Tokenizer = tokenizer;
-    }
-
-    public AcceptedCharactersInternal AcceptedCharacters { get; set; }
+    public required AcceptedCharactersInternal AcceptedCharacters { get; init; }
+    public required Func<string, IEnumerable<Syntax.InternalSyntax.SyntaxToken>> Tokenizer { get; init; }
 
-    public Func<string, IEnumerable<Syntax.InternalSyntax.SyntaxToken>> Tokenizer { get; set; }
-
-    public static SpanEditHandler CreateDefault()
```

Comment:
When refactoring, I found the implicit `AcceptedCharactersInternal` values confusing, so I removed this overload.